source/dub/compilers/compiler.d

```diff
@@ -124,18 +126,63 @@ interface Compiler {
 		This method should be used by `Compiler` implementations to invoke the
 		compiler or linker binary.
 	*/
-	protected final void invokeTool(string[] args, void delegate(int, string) output_callback, string[string] env = null)
+	protected final void invokeTool(string[] args, void delegate(int, string) output_callback, string[string] env = null, Duration timeout = Duration.max)
 	{
 		import std.string;
 
+		string[] timeouted(string[] args) @safe pure
+		{
+			if (timeout)
+			{
+				import std.conv : to;
+				return ["timeout", timeout.total!"seconds".to!string] ~ args;
+			}
+			return args;
+		}
+
 		int status;
 		if (output_callback) {
-			auto result = executeShell(escapeShellCommand(args), env);
+			auto result = executeShell(escapeShellCommand(timeouted(args)), env); // TODO: avoid using timeoutArgs
```

Comment:
Oh, I see that's what this does already. What's with the other branch, though?

Reply:
What branch?

Reply:
The `else` part of `if (output_callback) {`.